tap_pardot/client.py

```diff
@@ -54,12 +60,11 @@ def login(self):
             params={"format": "json"},
         )
 
-        # This will only work if they use HTTP codes. Handling Pardot
-        # errors below.
-        response.raise_for_status()
+        # 5xx errors should be retried
+        if response.status_code >= 500:
+            raise Pardot5xxError()
```

Comment:
I think we still want to maybe raise_for_status after the `_check_error` call below, to catch any statuses that we don't know about yet.